playbooks/troubleshooting/troubleshooting_guide.adoc

```diff
@@ -0,0 +1,719 @@
+kubernetes        172.50.0.1       <none>        443/TCP,53/UDP,53/TCP   <none>                    23h
+router            172.50.49.239    <none>        80/TCP                  router=router             20h
+----
+
+==== Solution
+
+This can be caused during the setup of the registry. If you have a
+change which triggers a re-ip of the docker-registry container (such as
+undeploy/redeploy) the old registry IP may be "stuck" in the
+configuration. When you recreate the service associated to the internal
+registry it will receive a new IP address. OpenShift masters do not
+automatically detect that change. Usually restarting the
+*atomic-openshift-master-api* service will fix the problem
+
+----
+systemctl restart atomic-openshift-master-api
+----
+
+=== OpenShift build error: failed to push image while using NFS persistent storage
+
+During the deployment of an application you see
+
+----
+Build error: Failed to push image. Response from registry is: Received unexpected HTTP status: 500 Internal Server Error
+----
+
+Check the host where the registry pod is supposed to be deployed:
+
+----
+[root@master00 ~]# oc get pods --all-namespaces
+NAMESPACE   NAME                      READY     STATUS      RESTARTS   AGE
+default     docker-registry-2-n8d21   1/1       Running     0          14h
+default     docker-registry-2-rlqzt   1/1       Running     1          15h
+default     router-1-47xfi            1/1       Running     2          15h
+default     router-1-vuw38            1/1       Running     2          15h
+
+[root@master00 ~]# oc describe pod docker-registry-2-n8d21
+Name:                docker-registry-2-n8d21
+Namespace:            default
+Image(s):            registry.access.redhat.com/openshift3/ose-docker-registry:v3.1.1.6
+Node:                infra01.ose.example.com/192.168.200.61
+----
+
+Check to see if the mount point exists on the node (in this case
+**infra01**)
+
+----
+[root@infra01 ~]# mount \|grep origin
+(rw,relatime,rootcontext="system_u:object_r:svirt_sandbox_file_t:s0:c0,c1",seclabel)
+192.168.200.99:/storage/vms/ose_nfs/stratus on /var/lib/origin/openshift.local.volumes/pods/06f81440-e64b-11e5-9d5e-0800270462ed/volumes/kubernetes.io~nfs/docker-registry-storage type nfs4 (rw,relatime,vers=4.0,rsize=1048576,wsize=1048576,namlen=255,hard,proto=tcp,port=0,timeo=600,retrans=2,sec=sys,clientaddr=192.168.200.61,local_lock=none,addr=192.168.200.99)
+----
+
+If the volume is mounted on the registry host, it is not a firewall, or
+nfs-server configuration issue.
+
+
+==== Solution
+
+It is likely that SELinux is blocking access to NFS from within the
+docker container. Check that the proper boolean is set:
+
+----
+[root@infra01 ~]# getsebool virt_use_nfs
+virt_use_nfs --> off
+----
+
+
+Set this boolean to on across any node that will host pods which may
+require NFS access (such as databases, registries etc):
+
+----
+setsebool -P virt_use_nfs=true
+----
+
+
+=== ha-registry with NFS persistent storage failed to push images
```

Comment:
Can we change the name of this issue to "Failure to push image to OpenShift's Registry when backed by shared storage"